feat: add operator info page

lcsfelix opened the pull request.

lcsfelix commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Create operator info screen](https://trello.com/c/0ukJnAUj/152-create-operator-info-screen)

lcsfelix commented:
I need to add error feedback before merging.

lcsfelix committed:
feat: add operator info page

feat: save/load operator info

feat: add formik switch component

feat: add input validation

fix: correct formik switch behaviour

fix: change infoCardEnabled to a radio input

style: move styles out of js

feat: add error feedback

RafaelTaranto commented on new-lamassu-admin/src/components/inputs/base/TextInput.js:
```diff
@@ -60,7 +110,7 @@ const TextInput = memo(
         }}
         InputLabelProps={{ className: classes.labelRoot }}
-        {...props}
+        {...R.omit(['className'], props)}
```
Why is className getting ommited and not used on the main parent? As it stands this prop does nothing.

lcsfelix replied:
`props.className` is omitted because it is already passed into the classNames object on line 86, which is used on line 100. If it was not omitted, the child component would receive **two** `className` props and there would be a conflict. This is visible on the Phone Number input where, if `props.className` is not omitted, the label has the wrong color when the input is empty.

RafaelTaranto replied:
Ahh, just saw that. Instead of using props.className you can desconstruct that.

`  ({ name, onChange, onBlur, value, error, suffix, large, ...props }) => {`
`  ({ name, onChange, onBlur, value, error, suffix, large, className, ...props }) => {`

and then you don't need to omit it.

lcsfelix replied:
Ah, good point. Changed in https://github.com/lamassu/lamassu-server/pull/354/commits/cb69441239bf2b56da02775aad37a5a16c28fe85

RafaelTaranto commented on new-lamassu-admin/src/pages/OperatorInfo/ContactInfo.js:
```diff
@@ -0,0 +1,306 @@
+  }
+
+  return (
+    <>
```
There's only one parent div, no need for the `<>`

RafaelTaranto commented on new-lamassu-admin/src/pages/OperatorInfo/ContactInfo.js:
```diff
@@ -0,0 +1,306 @@
+            website: Yup.string()
+              .url('Please enter a valid url')
+              .max(100, 'Too long'),
+            companyNumber: Yup.string().max(30, 'Too long')
```
Can you move this to objects outside of the html tags, easier to read that way.

RafaelTaranto commented on new-lamassu-admin/src/components/inputs/base/TextInput.js:
```diff
@@ -40,14 +81,23 @@ const TextInput = memo(
         id={name}
         onChange={onChange}
         onBlur={onBlur}
-        error={!!(touched[name] && errors[name])}
```
We need to remove this formik stuff from here, but it shouldn't break things. Errors should only show if it was already touched, can you add this logic to the formik specific text input?

lcsfelix committed:
fix: make textinput style guide compliant

fix: correct input spacing

fix: correct label color

fix: correct input width

fix: move formik stuff out of base TextInput

RafaelTaranto commented:
lgtm!
